### PR TITLE
Add Tuya soil sensor variant `_TZE284_aao3yzhs` and `_TZE284_sgabhwa6`

### LINKED ISSUE #3266: [Device Support Request] Tuya Soil Sensor (_TZE284_sgabhwa6 TS0601)
### Problem description

New Tuya Soil Sensor does not appear to have any clusters with temp/humidity readings after adding

https://www.aliexpress.us/item/3256806923087260.html

### Solution description

Device supported, reporting soil temp, humidity and battery level

### Screenshots/Video

<details><summary>Screenshots/Video</summary>

[Paste/upload your media here]

</details>

### Device signature

<details><summary>Device signature</summary>

```json
{
  "node_descriptor": "NodeDescriptor(logical_type=<LogicalType.EndDevice: 2>, complex_descriptor_available=0, user_descriptor_available=0, reserved=0, aps_flags=0, frequency_band=<FrequencyBand.Freq2400MHz: 8>, mac_capability_flags=<MACCapabilityFlags.AllocateAddress: 128>, manufacturer_code=4417, maximum_buffer_size=66, maximum_incoming_transfer_size=66, server_mask=10752, maximum_outgoing_transfer_size=66, descriptor_capability_field=<DescriptorCapability.NONE: 0>, *allocate_address=True, *is_alternate_pan_coordinator=False, *is_coordinator=False, *is_end_device=True, *is_full_function_device=False, *is_mains_powered=False, *is_receiver_on_when_idle=False, *is_router=False, *is_security_capable=False)",
  "endpoints": {
    "1": {
      "profile_id": "0x0104",
      "device_type": "0x0051",
      "input_clusters": [
        "0x0000",
        "0x0004",
        "0x0005",
        "0xed00",
        "0xef00"
      ],
      "output_clusters": [
        "0x000a",
        "0x0019"
      ]
    }
  },
  "manufacturer": "_TZE284_sgabhwa6",
  "model": "TS0601",
  "class": "zigpy.device.Device"
}
```

</details>

### Diagnostic information

<details><summary>Diagnostic information</summary>

```json
{
  "home_assistant": {
    "installation_type": "Home Assistant Container",
    "version": "2024.7.3",
    "dev": false,
    "hassio": false,
    "virtualenv": false,
    "python_version": "3.12.4",
    "docker": true,
    "arch": "x86_64",
    "timezone": "America/New_York",
    "os_name": "Linux",
    "os_version": "3.10.0-957.27.2.el7.x86_64",
    "run_as_root": true
  },
  "custom_components": {
    "sp108e_ws2815": {
      "documentation": "https://www.home-assistant.io/integrations/sp108e_ws2815",
      "version": "0.2.0",
      "requirements": []
    },
    "localtuya": {
      "documentation": "https://github.com/rospogrigio/localtuya/",
      "version": "5.2.1",
      "requirements": []
    },
    "eufy_security": {
      "documentation": "https://github.com/fuatakgun/eufy_security",
      "version": "7.7.4",
      "requirements": [
        "websocket-client==1.4.2",
        "aiortsp==1.3.7"
      ]
    },
    "hacs": {
      "documentation": "https://hacs.xyz/docs/configuration/start",
      "version": "1.34.0",
      "requirements": [
        "aiogithubapi>=22.10.1"
      ]
    },
    "bhyve": {
      "documentation": "https://github.com/sebr/bhyve-home-assistant/blob/main/README.md",
      "version": "3.2.4",
      "requirements": []
    },
    "openweathermaphistory": {
      "documentation": "https://github.com/petergridge/openweathermaphistory",
      "version": "2.0.12",
      "requirements": [
        "pyowm==3.2.0"
      ]
    }
  },
  "integration_manifest": {
    "domain": "zha",
    "name": "Zigbee Home Automation",
    "after_dependencies": [
      "onboarding",
      "usb"
    ],
    "codeowners": [
      "dmulcahey",
      "adminiuga",
      "puddly",
      "TheJulianJES"
    ],
    "config_flow": true,
    "dependencies": [
      "file_upload"
    ],
    "documentation": "https://www.home-assistant.io/integrations/zha",
    "iot_class": "local_polling",
    "loggers": [
      "aiosqlite",
      "bellows",
      "crccheck",
      "pure_pcapy3",
      "zhaquirks",
      "zigpy",
      "zigpy_deconz",
      "zigpy_xbee",
      "zigpy_zigate",
      "zigpy_znp",
      "universal_silabs_flasher"
    ],
    "requirements": [
      "bellows==0.39.1",
      "pyserial==3.5",
      "zha-quirks==0.0.117",
      "zigpy-deconz==0.23.2",
      "zigpy==0.64.1",
      "zigpy-xbee==0.20.1",
      "zigpy-zigate==0.12.1",
      "zigpy-znp==0.12.2",
      "universal-silabs-flasher==0.0.20",
      "pyserial-asyncio-fast==0.11"
    ],
    "usb": [
      {
        "vid": "10C4",
        "pid": "EA60",
        "description": "*2652*",
        "known_devices": [
          "slae.sh cc2652rb stick"
        ]
      },
      {
        "vid": "10C4",
        "pid": "EA60",
        "description": "*slzb-07*",
        "known_devices": [
          "smlight slzb-07"
        ]
      },
      {
        "vid": "1A86",
        "pid": "55D4",
        "description": "*sonoff*plus*",
        "known_devices": [
          "sonoff zigbee dongle plus v2"
        ]
      },
      {
        "vid": "10C4",
        "pid": "EA60",
        "description": "*sonoff*plus*",
        "known_devices": [
          "sonoff zigbee dongle plus"
        ]
      },
      {
        "vid": "10C4",
        "pid": "EA60",
        "description": "*tubeszb*",
        "known_devices": [
          "TubesZB Coordinator"
        ]
      },
      {
        "vid": "1A86",
        "pid": "7523",
        "description": "*tubeszb*",
        "known_devices": [
          "TubesZB Coordinator"
        ]
      },
      {
        "vid": "1A86",
        "pid": "7523",
        "description": "*zigstar*",
        "known_devices": [
          "ZigStar Coordinators"
        ]
      },
      {
        "vid": "1CF1",
        "pid": "0030",
        "description": "*conbee*",
        "known_devices": [
          "Conbee II"
        ]
      },
      {
        "vid": "0403",
        "pid": "6015",
        "description": "*conbee*",
        "known_devices": [
          "Conbee III"
        ]
      },
      {
        "vid": "10C4",
        "pid": "8A2A",
        "description": "*zigbee*",
        "known_devices": [
          "Nortek HUSBZB-1"
        ]
      },
      {
        "vid": "0403",
        "pid": "6015",
        "description": "*zigate*",
        "known_devices": [
          "ZiGate+"
        ]
      },
      {
        "vid": "10C4",
        "pid": "EA60",
        "description": "*zigate*",
        "known_devices": [
          "ZiGate"
        ]
      },
      {
        "vid": "10C4",
        "pid": "8B34",
        "description": "*bv 2010/10*",
        "known_devices": [
          "Bitron Video AV2010/10"
        ]
      }
    ],
    "zeroconf": [
      {
        "type": "_esphomelib._tcp.local.",
        "name": "tube*"
      },
      {
        "type": "_zigate-zigbee-gateway._tcp.local.",
        "name": "*zigate*"
      },
      {
        "type": "_zigstar_gw._tcp.local.",
        "name": "*zigstar*"
      },
      {
        "type": "_uzg-01._tcp.local.",
        "name": "uzg-01*"
      },
      {
        "type": "_slzb-06._tcp.local.",
        "name": "slzb-06*"
      },
      {
        "type": "_xzg._tcp.local.",
        "name": "xzg*"
      },
      {
        "type": "_czc._tcp.local.",
        "name": "czc*"
      }
    ],
    "is_built_in": true
  },
  "setup_times": {
    "null": {
      "setup": 0.0001849830150604248
    },
    "3d2c866ed2de507c75ae2277a654467a": {
      "wait_import_platforms": -0.0005826279520988464,
      "config_entry_setup": 29.70109256543219
    }
  },
  "data": {
    "ieee": "**REDACTED**",
    "nwk": 46680,
    "manufacturer": "_TZE284_sgabhwa6",
    "model": "TS0601",
    "name": "_TZE284_sgabhwa6 TS0601",
    "quirk_applied": false,
    "quirk_class": "zigpy.device.Device",
    "quirk_id": null,
    "manufacturer_code": 4417,
    "power_source": "Battery or Unknown",
    "lqi": 51,
    "rssi": null,
    "last_seen": "2024-07-20T16:23:40",
    "available": true,
    "device_type": "EndDevice",
    "signature": {
      "node_descriptor": "NodeDescriptor(logical_type=<LogicalType.EndDevice: 2>, complex_descriptor_available=0, user_descriptor_available=0, reserved=0, aps_flags=0, frequency_band=<FrequencyBand.Freq2400MHz: 8>, mac_capability_flags=<MACCapabilityFlags.AllocateAddress: 128>, manufacturer_code=4417, maximum_buffer_size=66, maximum_incoming_transfer_size=66, server_mask=10752, maximum_outgoing_transfer_size=66, descriptor_capability_field=<DescriptorCapability.NONE: 0>, *allocate_address=True, *is_alternate_pan_coordinator=False, *is_coordinator=False, *is_end_device=True, *is_full_function_device=False, *is_mains_powered=False, *is_receiver_on_when_idle=False, *is_router=False, *is_security_capable=False)",
      "endpoints": {
        "1": {
          "profile_id": "0x0104",
          "device_type": "0x0051",
          "input_clusters": [
            "0x0000",
            "0x0004",
            "0x0005",
            "0xed00",
            "0xef00"
          ],
          "output_clusters": [
            "0x000a",
            "0x0019"
          ]
        }
      },
      "manufacturer": "_TZE284_sgabhwa6",
      "model": "TS0601"
    },
    "active_coordinator": false,
    "entities": [
      {
        "entity_id": "update.green_soil_sensor_deck_pot_firmware",
        "name": "_TZE284_sgabhwa6 TS0601"
      }
    ],
    "neighbors": [],
    "routes": [],
    "endpoint_names": [
      {
        "name": "SMART_PLUG"
      }
    ],
    "user_given_name": "Green Soil Sensor - Deck Pot",
    "device_reg_id": "e5903ec6af30fc40bf73dc154cb81a68",
    "area_id": "backyard",
    "cluster_details": {
      "1": {
        "device_type": {
          "name": "SMART_PLUG",
          "id": 81
        },
        "profile_id": 260,
        "in_clusters": {
          "0x0004": {
            "endpoint_attribute": "groups",
            "attributes": {
              "0xfffd": {
                "attribute_name": "cluster_revision",
                "value": 2
              },
              "0x0000": {
                "attribute_name": "name_support",
                "value": 0
              }
            },
            "unsupported_attributes": {
              "0xfffe": {
                "attribute_name": "reporting_status"
              }
            }
          },
          "0x0005": {
            "endpoint_attribute": "scenes",
            "attributes": {
              "0xfffd": {
                "attribute_name": "cluster_revision",
                "value": 2
              },
              "0x0000": {
                "attribute_name": "count",
                "value": 0
              },
              "0x0002": {
                "attribute_name": "current_group",
                "value": 0
              },
              "0x0001": {
                "attribute_name": "current_scene",
                "value": 0
              },
              "0x0004": {
                "attribute_name": "name_support",
                "value": 0
              },
              "0x0003": {
                "attribute_name": "scene_valid",
                "value": 0
              }
            },
            "unsupported_attributes": {
              "0xfffe": {
                "attribute_name": "reporting_status"
              },
              "0x0005": {
                "attribute_name": "last_configured_by"
              }
            }
          },
          "0xef00": {
            "endpoint_attribute": null,
            "attributes": {},
            "unsupported_attributes": {}
          },
          "0x0000": {
            "endpoint_attribute": "basic",
            "attributes": {
              "0x0001": {
                "attribute_name": "app_version",
                "value": 77
              },
              "0x0004": {
                "attribute_name": "manufacturer",
                "value": "_TZE284_sgabhwa6"
              },
              "0x0005": {
                "attribute_name": "model",
                "value": "TS0601"
              }
            },
            "unsupported_attributes": {}
          },
          "0xed00": {
            "endpoint_attribute": null,
            "attributes": {},
            "unsupported_attributes": {}
          }
        },
        "out_clusters": {
          "0x0019": {
            "endpoint_attribute": "ota",
            "attributes": {
              "0x0002": {
                "attribute_name": "current_file_version",
                "value": 77
              }
            },
            "unsupported_attributes": {}
          },
          "0x000a": {
            "endpoint_attribute": "time",
            "attributes": {
              "0xfffd": {
                "attribute_name": "cluster_revision",
                "value": 1
              }
            },
            "unsupported_attributes": {
              "0x0000": {
                "attribute_name": "time"
              },
              "0x0001": {
                "attribute_name": "time_status"
              },
              "0x0002": {
                "attribute_name": "time_zone"
              },
              "0x0003": {
                "attribute_name": "dst_start"
              },
              "0x0004": {
                "attribute_name": "dst_end"
              },
              "0x0005": {
                "attribute_name": "dst_shift"
              },
              "0x0006": {
                "attribute_name": "standard_time"
              },
              "0x0007": {
                "attribute_name": "local_time"
              },
              "0x0008": {
                "attribute_name": "last_set_time"
              },
              "0x0009": {
                "attribute_name": "valid_until_time"
              },
              "0xfffe": {
                "attribute_name": "reporting_status"
              }
            }
          }
        }
      }
    }
  }
}
```

</details>

### Logs

<details><summary>Logs</summary>

```python
[Paste the logs here]
```

</details>

### Custom quirk

<details><summary>Custom quirk</summary>

```python
[Paste your custom quirk here]
```

</details>

### Additional information

_No response_

### LINKED ISSUE #3305: [Device Support Request] Plant Monitor SGS01Z
### Problem description

The Plant Monitor can be paired but doesn't show any entities 

It shows up as a TS0601 _TZE284_aao3yzhs

Link: https://www.aliexpress.com/item/1005007330089764.html

image
image

### Solution description

Add a custom quirk to display Soil moisture and temperature 

### Screenshots/Video

<details><summary>Screenshots/Video</summary>

[Paste/upload your media here]

</details>

### Device signature

<details><summary>Device signature</summary>

```json
{
  "node_descriptor": {
    "logical_type": 2,
    "complex_descriptor_available": 0,
    "user_descriptor_available": 0,
    "reserved": 0,
    "aps_flags": 0,
    "frequency_band": 8,
    "mac_capability_flags": 128,
    "manufacturer_code": 4417,
    "maximum_buffer_size": 66,
    "maximum_incoming_transfer_size": 66,
    "server_mask": 10752,
    "maximum_outgoing_transfer_size": 66,
    "descriptor_capability_field": 0
  },
  "endpoints": {
    "1": {
      "profile_id": "0x0104",
      "device_type": "0x0051",
      "input_clusters": [
        "0x0000",
        "0x0004",
        "0x0005",
        "0xed00",
        "0xef00"
      ],
      "output_clusters": [
        "0x000a",
        "0x0019"
      ]
    }
  },
  "manufacturer": "_TZE284_aao3yzhs",
  "model": "TS0601",
  "class": "zigpy.device.Device"
}
```

</details>

### Diagnostic information

<details><summary>Diagnostic information</summary>

```json
{
  "home_assistant": {
    "installation_type": "Home Assistant OS",
    "version": "2024.8.2",
    "dev": false,
    "hassio": true,
    "virtualenv": false,
    "python_version": "3.12.4",
    "docker": true,
    "arch": "x86_64",
    "timezone": "Europe/Amsterdam",
    "os_name": "Linux",
    "os_version": "6.6.44-haos",
    "supervisor": "2024.08.0",
    "host_os": "Home Assistant OS 13.0",
    "docker_version": "26.1.4",
    "chassis": "vm",
    "run_as_root": true
  },
  "custom_components": {
    "average": {
      "documentation": "https://github.com/Limych/ha-average",
      "version": "2.3.4",
      "requirements": []
    },
    "knmi": {
      "documentation": "https://github.com/golles/ha-knmi/blob/main/README.md",
      "version": "2.2.0",
      "requirements": []
    },
    "philips_airpurifier_coap": {
      "documentation": "https://github.com/kongo09/philips-airpurifier-coap",
      "version": "0.18.5",
      "requirements": [
        "aioairctrl==0.2.5",
        "getmac==0.9.4"
      ]
    },
    "hacs": {
      "documentation": "https://hacs.xyz/docs/configuration/start",
      "version": "1.34.0",
      "requirements": [
        "aiogithubapi>=22.10.1"
      ]
    }
  },
  "integration_manifest": {
    "domain": "zha",
    "name": "Zigbee Home Automation",
    "after_dependencies": [
      "onboarding",
      "usb"
    ],
    "codeowners": [
      "dmulcahey",
      "adminiuga",
      "puddly",
      "TheJulianJES"
    ],
    "config_flow": true,
    "dependencies": [
      "file_upload"
    ],
    "documentation": "https://www.home-assistant.io/integrations/zha",
    "iot_class": "local_polling",
    "loggers": [
      "aiosqlite",
      "bellows",
      "crccheck",
      "pure_pcapy3",
      "zhaquirks",
      "zigpy",
      "zigpy_deconz",
      "zigpy_xbee",
      "zigpy_zigate",
      "zigpy_znp",
      "zha",
      "universal_silabs_flasher"
    ],
    "requirements": [
      "universal-silabs-flasher==0.0.22",
      "zha==0.0.31"
    ],
    "usb": [
      {
        "vid": "10C4",
        "pid": "EA60",
        "description": "*2652*",
        "known_devices": [
          "slae.sh cc2652rb stick"
        ]
      },
      {
        "vid": "10C4",
        "pid": "EA60",
        "description": "*slzb-07*",
        "known_devices": [
          "smlight slzb-07"
        ]
      },
      {
        "vid": "1A86",
        "pid": "55D4",
        "description": "*sonoff*plus*",
        "known_devices": [
          "sonoff zigbee dongle plus v2"
        ]
      },
      {
        "vid": "10C4",
        "pid": "EA60",
        "description": "*sonoff*plus*",
        "known_devices": [
          "sonoff zigbee dongle plus"
        ]
      },
      {
        "vid": "10C4",
        "pid": "EA60",
        "description": "*tubeszb*",
        "known_devices": [
          "TubesZB Coordinator"
        ]
      },
      {
        "vid": "1A86",
        "pid": "7523",
        "description": "*tubeszb*",
        "known_devices": [
          "TubesZB Coordinator"
        ]
      },
      {
        "vid": "1A86",
        "pid": "7523",
        "description": "*zigstar*",
        "known_devices": [
          "ZigStar Coordinators"
        ]
      },
      {
        "vid": "1CF1",
        "pid": "0030",
        "description": "*conbee*",
        "known_devices": [
          "Conbee II"
        ]
      },
      {
        "vid": "0403",
        "pid": "6015",
        "description": "*conbee*",
        "known_devices": [
          "Conbee III"
        ]
      },
      {
        "vid": "10C4",
        "pid": "8A2A",
        "description": "*zigbee*",
        "known_devices": [
          "Nortek HUSBZB-1"
        ]
      },
      {
        "vid": "0403",
        "pid": "6015",
        "description": "*zigate*",
        "known_devices": [
          "ZiGate+"
        ]
      },
      {
        "vid": "10C4",
        "pid": "EA60",
        "description": "*zigate*",
        "known_devices": [
          "ZiGate"
        ]
      },
      {
        "vid": "10C4",
        "pid": "8B34",
        "description": "*bv 2010/10*",
        "known_devices": [
          "Bitron Video AV2010/10"
        ]
      }
    ],
    "zeroconf": [
      {
        "type": "_esphomelib._tcp.local.",
        "name": "tube*"
      },
      {
        "type": "_zigate-zigbee-gateway._tcp.local.",
        "name": "*zigate*"
      },
      {
        "type": "_zigstar_gw._tcp.local.",
        "name": "*zigstar*"
      },
      {
        "type": "_uzg-01._tcp.local.",
        "name": "uzg-01*"
      },
      {
        "type": "_slzb-06._tcp.local.",
        "name": "slzb-06*"
      },
      {
        "type": "_xzg._tcp.local.",
        "name": "xzg*"
      },
      {
        "type": "_czc._tcp.local.",
        "name": "czc*"
      }
    ],
    "is_built_in": true
  },
  "setup_times": {
    "null": {
      "setup": 7.1326969191432e-05
    },
    "eccd7b20eecde60a7a4b75548536550c": {
      "wait_import_platforms": -0.016558641975279897,
      "wait_base_component": -0.0007912219734862447,
      "config_entry_setup": 12.98326518398244
    }
  },
  "data": {
    "ieee": "**REDACTED**",
    "nwk": 32106,
    "manufacturer": "_TZE284_aao3yzhs",
    "model": "TS0601",
    "name": "_TZE284_aao3yzhs TS0601",
    "quirk_applied": false,
    "quirk_class": "zigpy.device.Device",
    "quirk_id": null,
    "manufacturer_code": 4417,
    "power_source": "Battery or Unknown",
    "lqi": 100,
    "rssi": -75,
    "last_seen": "2024-08-19T13:34:24",
    "available": false,
    "device_type": "EndDevice",
    "signature": {
      "node_descriptor": {
        "logical_type": 2,
        "complex_descriptor_available": 0,
        "user_descriptor_available": 0,
        "reserved": 0,
        "aps_flags": 0,
        "frequency_band": 8,
        "mac_capability_flags": 128,
        "manufacturer_code": 4417,
        "maximum_buffer_size": 66,
        "maximum_incoming_transfer_size": 66,
        "server_mask": 10752,
        "maximum_outgoing_transfer_size": 66,
        "descriptor_capability_field": 0
      },
      "endpoints": {
        "1": {
          "profile_id": "0x0104",
          "device_type": "0x0051",
          "input_clusters": [
            "0x0000",
            "0x0004",
            "0x0005",
            "0xed00",
            "0xef00"
          ],
          "output_clusters": [
            "0x000a",
            "0x0019"
          ]
        }
      },
      "manufacturer": "_TZE284_aao3yzhs",
      "model": "TS0601"
    },
    "active_coordinator": false,
    "entities": [
      {
        "entity_id": "update.tze284_aao3yzhs_ts0601_firmware",
        "name": "_TZE284_aao3yzhs TS0601"
      },
      {
        "entity_id": "update.plant_monitor_1_firmware",
        "name": "_TZE284_aao3yzhs TS0601"
      }
    ],
    "neighbors": [],
    "routes": [],
    "endpoint_names": [
      {
        "name": "SMART_PLUG"
      }
    ],
    "user_given_name": "Plant Monitor 1",
    "device_reg_id": "dd581646b6c8df6f29d816b4d4835118",
    "area_id": "bedroom",
    "cluster_details": {
      "1": {
        "device_type": {
          "name": "SMART_PLUG",
          "id": 81
        },
        "profile_id": 260,
        "in_clusters": {
          "0x0004": {
            "endpoint_attribute": "groups",
            "attributes": {},
            "unsupported_attributes": {}
          },
          "0x0005": {
            "endpoint_attribute": "scenes",
            "attributes": {},
            "unsupported_attributes": {}
          },
          "0xef00": {
            "endpoint_attribute": null,
            "attributes": {},
            "unsupported_attributes": {}
          },
          "0x0000": {
            "endpoint_attribute": "basic",
            "attributes": {
              "0x0004": {
                "attribute_name": "manufacturer",
                "value": "_TZE284_aao3yzhs"
              },
              "0x0005": {
                "attribute_name": "model",
                "value": "TS0601"
              }
            },
            "unsupported_attributes": {}
          },
          "0xed00": {
            "endpoint_attribute": null,
            "attributes": {},
            "unsupported_attributes": {}
          }
        },
        "out_clusters": {
          "0x0019": {
            "endpoint_attribute": "ota",
            "attributes": {
              "0x0002": {
                "attribute_name": "current_file_version",
                "value": 77
              }
            },
            "unsupported_attributes": {}
          },
          "0x000a": {
            "endpoint_attribute": "time",
            "attributes": {},
            "unsupported_attributes": {}
          }
        }
      }
    }
  }
}
```

</details>

### Logs

<details><summary>Logs</summary>

```python
[Paste the logs here]
```

</details>

### Custom quirk

<details><summary>Custom quirk</summary>

```python
[Paste your custom quirk here]
```

</details>

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by prairiesnpr**
## Proposed change

Add support for TUYA TS0601 _TZE284_aao3yzhs

Also adds support for TS0601_TZE284_sgabhwa6.  @GeoffWy confirms suitability for TS0601_TZE284_sgabhwa6.

## Additional information

Fixes: https://github.com/zigpy/zha-device-handlers/issues/3305
Fixes: https://github.com/zigpy/zha-device-handlers/issues/3266

## Checklist

- [X] The changes are tested and work correctly
- [X] `pre-commit` checks pass / the code has been formatted using Black
- [ ] Tests have been added to verify that the new code works

### CODE REVIEW

**Comment by GeoffWy** [comment]
This quirk also works for Tuya soil sensor variant _TZE284_sgabhwa6

File: zhaquirks/tuya/ts0601_sensor.py [hunk 1]
```diff
@@ -407,3 +407,79 @@ class TuyaSoilSensor(CustomDevice):
...
+
+    data_point_handlers = {
+        3: "_dp_2_attr_update",
+        5: "_dp_2_attr_update",
+        15: "_dp_2_attr_update",
+    }
+
+
+class TuyaSoilSensorVar02(CustomDevice):
+    """Tuya temp and humidity sensor (variation 05)."""
```
Comment (TheJulianJES) [inline zhaquirks/tuya/ts0601_sensor.py:441]:
```suggestion
    """Tuya soil sensor (variation 2)."""
```
The other docstring is also wrong, can't add a suggestion for that here though. Feel free to PR the change for `TuyaSoilSensor`. It still says "temp and humidity sensor".

EDIT: Actually, this is a temp/humidity sensor. So, it's fine.

Comment (TheJulianJES) [reply zhaquirks/tuya/ts0601_sensor.py:441]: The numbering seems a bit wrong overall. Feel free to improve it in a future PR.

**Review by TheJulianJES — APPROVED** [review]
Thanks!

**Comment by skypyxis** [comment]
I have a _TZE284_sgabhwa6 and I confirm I can get the temperature and moisture readings.
However, the battery level isn't being detected, and the moisture level seems quite low. Still, this is definitely promising

image

**Comment by hairyfred** [comment]
Got a TZE284_aao3yzhs so fingers crossed for when this releases :)

**Comment by GeoffWy** [comment]
I agree the soil moisture reading seems quite low, but I have confirmed it does read 100% (or near enough) when the probes are fully immersed in water, so I believe the readings are being scaled correctly.

My battery level isn't being detected either, but my experience has been that some sensors take 24 hours to start producing battery level, so still waiting in hope.

It would be nice to change the battery type to 3AA instead of 2AAA.

**Comment by prairiesnpr** [comment]
Quickly looking at the z2m convertor, I doubt the battery will work for _TZE284_sgabhwa6. Looks like it uses dp 115 for battery where the _TZE284_aao3yzhs uses 15.

**Comment by BonesJD** [comment]
Hey sorry I'm a bit confused about the status of this. Has support been implemented yet or not?
I just bought a "TS0601 by _TZE284_aao3yzhs" and connected through ZHA but I'm not seeing any sensor readings.
image
Am I doing something wrong?

**Comment by prairiesnpr** [comment]
It's been in since HA 2024.9, if you are on the latest HA and it's still not showing, I'd open an incident with your signature, diagnostics, and logs.

**Comment by BonesJD** [comment]
> It's been in since HA 2024.9, if you are on the latest HA and it's still not showing, I'd open an incident with your signature, diagnostics, and logs.

Sorry that was a simple fix, I should have checked the version first. I just installed home assistant from the CasaOS store yesterday so I assumed it would be the latest version but it was 2024.5 for some reason. I updated to the latest and it's working correctly now. Thanks!

**Comment by Trilis29** [comment]
Hello guys, I have _TZE284_aao3yzhs but Soil moisture mostly always 100% or nearby. Should I add some quirk, because I see this merged but not correctly showing. I am on latest HA 2025.3.3

**Comment by tablatronix** [comment]
I only have 1 of these but its Soil Moisture is always way high , about 2x what other sensors read, tells me a pot is 74% which is impossible. temp seems ok
Could this be a handler issue or just a bad unit ?

**Comment by putinet** [comment]
MODELS_INFO: [
            ("_TZE284_k7p2q5d9", "TS0601"),


[ts0601_sensor.py](https://github.com/user-attachments/files/23973591/ts0601_sensor.py)


"""Tuya Soil Sensor TZE284 Custom Quirk - Clean Version (Temp, Soil, Battery only)."""

from typing import Any

from zigpy.profiles import zha
from zigpy.quirks import CustomDevice
from zigpy.zcl.clusters.general import (
    Basic,
    Groups,
    Ota,
    Scenes,
    Time,
    Identify,
    PowerConfiguration,
)
from zigpy.zcl.clusters.lightlink import LightLink
from zigpy.zcl.clusters.measurement import (
    SoilMoisture,
    TemperatureMeasurement,
)

from zhaquirks.const import (
    DEVICE_TYPE,
    ENDPOINTS,
    INPUT_CLUSTERS,
    MODELS_INFO,
    OUTPUT_CLUSTERS,
    PROFILE_ID,
    SKIP_CONFIGURATION,
)
from zhaquirks.tuya import TuyaLocalCluster, TuyaPowerConfigurationCluster2AAA
from zhaquirks.tuya.mcu import DPToAttributeMapping, TuyaMCUCluster


# ----------------------------------------------------------------------------
# Clases Auxiliares
# ----------------------------------------------------------------------------

class TuyaTemperatureMeasurement(TemperatureMeasurement, TuyaLocalCluster):
    """Tuya local TemperatureMeasurement cluster."""


class TuyaSoilMoisture(SoilMoisture, TuyaLocalCluster):
    """Tuya local SoilMoisture cluster."""


class SoilManufCluster_Clean(TuyaMCUCluster):
    """
    Manejador de Data Points limpio.
    Solo incluye Temperatura, Humedad del Suelo y Batería.
    """
    
    dp_to_attribute: dict[int, DPToAttributeMapping] = {
        # --- TEMPERATURA ---
        5: DPToAttributeMapping(
            TuyaTemperatureMeasurement.ep_attribute,
            "measured_value",
            converter=lambda x: x * 10,
        ),
        
        # --- HUMEDAD DEL SUELO ---
        3: DPToAttributeMapping(
            TuyaSoilMoisture.ep_attribute,
            "measured_value",
            converter=lambda x: x * 100,
        ),

        # --- BATERÍA ---
        # Mantenemos la lógica que te funcionó para dar el 100%
        15: DPToAttributeMapping(
            TuyaPowerConfigurationCluster2AAA.ep_attribute,
            "battery_percentage_remaining",
            converter=lambda x: x * 2 if x > 10 else x * 100, 
        ),
        # Canales alternativos de batería por seguridad
        4: DPToAttributeMapping(
            TuyaPowerConfigurationCluster2AAA.ep_attribute,
            "battery_percentage_remaining",
            converter=lambda x: x * 2 if x > 10 else x * 100,
        ),
        14: DPToAttributeMapping(
            TuyaPowerConfigurationCluster2AAA.ep_attribute,
            "battery_percentage_remaining",
            converter=lambda x: x * 2 if x > 10 else x * 100,
        ),
    }

    # Solo escuchamos los canales necesarios
    data_point_handlers = {
        3: "_dp_2_attr_update",
        4: "_dp_2_attr_update",
        5: "_dp_2_attr_update",
        14: "_dp_2_attr_update",
        15: "_dp_2_attr_update",
    }


# ----------------------------------------------------------------------------
# Definición del Dispositivo
# ----------------------------------------------------------------------------

class TuyaSoilSensor_TZE284_Switch_Style(CustomDevice):
    """Soporte optimizado para TZE284 (Solo funciones básicas)."""

    signature = {
        MODELS_INFO: [
            ("_TZE284_k7p2q5d9", "TS0601"),
        ],
        ENDPOINTS: {
            1: {
                PROFILE_ID: zha.PROFILE_ID,          
                DEVICE_TYPE: zha.DeviceType.ON_OFF_SWITCH, 
                INPUT_CLUSTERS: [
                    Basic.cluster_id,                
                    PowerConfiguration.cluster_id,   
                    Identify.cluster_id,             
                    TuyaMCUCluster.cluster_id,       
                ],
                OUTPUT_CLUSTERS: [
                    LightLink.cluster_id,            
                    TuyaMCUCluster.cluster_id,       
                ],
            }
        },
    }

    replacement = {
        SKIP_CONFIGURATION: True,
        ENDPOINTS: {
            1: {
                DEVICE_TYPE: zha.DeviceType.TEMPERATURE_SENSOR,
                INPUT_CLUSTERS: [
                    Basic.cluster_id,
                    Identify.cluster_id,
                    
                    # Nuestro manejador de datos limpio
                    SoilManufCluster_Clean,       
                    
                    # Entidades (Solo las que funcionan)
                    TuyaTemperatureMeasurement,         
                    TuyaSoilMoisture,                   
                    TuyaPowerConfigurationCluster2AAA,  
                ],
                OUTPUT_CLUSTERS: [Ota.cluster_id, Time.cluster_id],
            }
        }, 
    }

**Pull request merged**